app/jobs/delete_user_job.rb

```diff
@@ -2,15 +2,74 @@ class DeleteUserJob < ApplicationJob
   queue_as :default
   queue_with_priority PRIORITIES.fetch(:profile_deletion)
 
+  ASSOCIATIONS = {
+    approved_ownership_requests: :destroy,
+    closed_ownership_calls: :destroy,
+    closed_ownership_requests: :destroy,
+    oidc_pending_trusted_publishers: :destroy,
+    ownership_calls: :destroy,
+    ownership_requests: :destroy,
+    ownerships: :destroy,
+    subscriptions: :destroy,
+    unconfirmed_ownerships: :destroy,
+    web_hooks: :destroy,
+    webauthn_credentials: :destroy,
+    webauthn_verification: :destroy,
+
+    api_keys: nil,
+    audits: nil,
+    deletions: nil,
+    oidc_api_key_roles: nil,
+    pushed_versions: nil
+  }.freeze
+
   def perform(user:)
```

Comment:
I would expect this logic to be part of model itself and in background job just got called something like `user.soft_delete`. Btw. would it make sense to rename this job to `SoftDeleteUserJob`?

Reply:
I agree, this solves the concern about a split canonical information about deletes.